src/modules/PortAudio/PortAudioWriter.cpp

```diff
@@ -203,12 +240,36 @@ bool PortAudioWriter::open()
 
 bool PortAudioWriter::openStream()
 {
+#ifdef Q_OS_MACOS
+	bool bitPerfectMode = sets().getBool("BitPerfect");
+	if (bitPerfectMode && (outputDevice.isEmpty() || outputDevice == "Default"))
+	{
+		int chn = getParam("chn").toInt();
+		outputParameters.device = PortAudioCommon::getDeviceIndexForOutput(outputDevice, chn);
+	}
+#endif
 	PaStream *newStream = nullptr;
 	if (Pa_OpenStream(&newStream, nullptr, &outputParameters, sample_rate, 0, paDitherOff, nullptr, nullptr) == paNoError)
 	{
 		stream = newStream;
 		outputLatency = Pa_GetStreamInfo(stream)->outputLatency;
 		modParam("delay", outputLatency);
+#ifdef Q_OS_MACOS
+		modParam("rate", Pa_GetStreamInfo(stream)->sampleRate);
```

Comment:
This parameter is set by `AudioThr` and shouldn't be changed here. Maybe use class member variable if it is needed for `PortAudioWriter::processParams`? And shouldn't it be inside `if (bitPerfectMode)`?

Reply:
Again, I'll have to re-test to see if I can remember why I added the statement.